apps/managedidentity/managedidentity.go

```diff
@@ -204,28 +292,183 @@ func (client Client) getTokenForRequest(req *http.Request) (accesstokens.TokenRe
 			Err:  fmt.Errorf("http call(%s)(%s) error: reply status code was %d", req.URL.String(), req.Method, resp.StatusCode),
 		}
 	}
-	var r accesstokens.TokenResponse
+
 	err = json.Unmarshal(responseBytes, &r)
 	return r, err
 }
 
-// Acquires tokens from the configured managed identity on an azure resource.
-//
-// Resource: scopes application is requesting access to
-// Options: [WithClaims]
-func (client Client) AcquireToken(ctx context.Context, resource string, options ...AcquireTokenOption) (base.AuthResult, error) {
-	o := AcquireTokenOptions{}
+func fileExists(path string) bool {
+	if _, err := os.Stat(path); err != nil {
+		if os.IsNotExist(err) {
+			return false
+		}
+	}
+	return true
+}
 
-	for _, option := range options {
-		option(&o)
+func createIMDSAuthRequest(ctx context.Context, id ID, resource string) (*http.Request, error) {
+	var msiEndpoint *url.URL
+	msiEndpoint, err := url.Parse(imdsEndpoint)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't parse %q: %s", imdsEndpoint, err)
+	}
+	msiParameters := msiEndpoint.Query()
+	msiParameters.Set(apiVersionQueryParameterName, imdsAPIVersion)
+	resource = strings.TrimSuffix(resource, "/.default")
+	msiParameters.Set(resourceQueryParameterName, resource)
+
+	switch t := id.(type) {
+	case UserAssignedClientID:
+		msiParameters.Set(miQueryParameterClientId, string(t))
+	case UserAssignedResourceID:
+		msiParameters.Set(miQueryParameterResourceId, string(t))
+	case UserAssignedObjectID:
+		msiParameters.Set(miQueryParameterObjectId, string(t))
+	case systemAssignedValue: // not adding anything
+	default:
+		return nil, fmt.Errorf("unsupported type %T", id)
 	}
-	req, err := createIMDSAuthRequest(ctx, client.miType, resource, o.claims)
+
+	msiEndpoint.RawQuery = msiParameters.Encode()
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, msiEndpoint.String(), nil)
 	if err != nil {
-		return base.AuthResult{}, err
+		return nil, fmt.Errorf("error creating http request %s", err)
+	}
+	req.Header.Set(metaHTTPHeaderName, "true")
+	return req, nil
+}
+
+func createAzureArcAuthRequest(ctx context.Context, id ID, resource string) (*http.Request, error) {
+	identityEndpoint, _, _ := getAndValidateAzureArcEnvVars()
+	var msiEndpoint *url.URL
+
+	if identityEndpoint == "" {
```

Comment:
This will never be empty, since the method getAndValidateAzureArcEnvVars will return it

Reply:
This is to actually get the endpoint as in the GetSource it is more to just get what source we want, we don't set the endpoint there, while here we are getting the endpoint to use

Reply:
Nice spot thanks! Updated!